platformio.ini

```diff
@@ -120,6 +145,7 @@ board = lolin_s3
 board_build.f_cpu = 240000000L
 board_upload.flash_size = 16MB
 board_build.partitions = esp32_partition_16M.csv
+board_build.extra_flags = -DBOARD_HAS_PSRAM
```

Comment:
Again this overwrites the extra_flags with USB settings, lolinS3 has psram, no need to set it here.

Reply:
this target just builds the binary, none of the USB settings are needed. eventually I'm going to remove the 'lolin_s3' and replace it with a generic 'esp32' for all boards.

Reply:
We had this before, the serial console on S3 is not working if the USB extra flags are not set.  This board profile already has PSRAM set, so this entry is bad, it does not affect PSRAM, but clears all other flags.

Reply:
yes I remember, but I can't reproduce when testing on two different S3s. Wait until I've modified the board's and try again.

Reply:
I remember you have a S3 with 2 USB, one native and one with usb/serial chip. The native does not work with serial console if you modify the extra-flags.
Settings in the board file for lolin-s3:
```
    "extra_flags": [
      "-DARDUINO_LOLIN_S3",
      "-DBOARD_HAS_PSRAM",
      "-DARDUINO_USB_MODE=1",
      "-DARDUINO_USB_CDC_ON_BOOT=1",
      "-DARDUINO_RUNNING_CORE=1",
      "-DARDUINO_EVENT_RUNNING_CORE=1"
    ],
```